### PR TITLE
Make concurrent move_base_flex melodic compatible

### PR DESCRIPTION
**Pull request by jabbenseth**
The main changes were
1) somehow the signature of actionlib::ActionServer doesn't support goal and cancel callbacks with references anymore (although I have not found out how it worked before)

2) Use tf2 if running on melodic: costmap2d switched to tf2 with melodic. Depending on the ros-distro different typedefs are used and in mbf_utility the pose transformation is done with tf2 for melodic

### CODE REVIEW

**Review by corot — APPROVED** [review]
Thanks a lot! I was trying to find some time to install melodic and fix the compilation error on action servers (due to new boost version).

**Comment by jabbenseth** [comment]
I'm not totally sure about the fix though. The passing is now by value, but since it is a handle consisting only of shared pointers and the copy constructor / `GoalHandle::operator=` does not create a deepcopy so it should be fine

File: mbf_abstract_nav/CMakeLists.txt [hunk 1]
```diff
@@ -50,6 +50,12 @@ catkin_package(
   DEPENDS Boost
 )
 
+if("$ENV{ROS_DISTRO}" STREQUAL "melodic")
```
Comment (spuetz) [inline mbf_abstract_nav/CMakeLists.txt]: We should make it the other way around. So for `kinetic` and `luna` the define `USE_OLD_TF`. New ros distros would be autometically included.

**Review by spuetz — CHANGES_REQUESTED** [review]
Thanks for the pull request! Just one little thing I would chnage. All the rest looks good to me! @corot What do you think?

File: mbf_abstract_nav/CMakeLists.txt (hunk 1 above)
Comment (corot) [reply mbf_abstract_nav/CMakeLists.txt]: sounds reasonable, yes.  always looking forward!

For the change on the action servers, remove the reference is the first thing I wanted to try, so... I feel confirmed.

**Comment by jabbenseth** [comment]
not yet tested on kinetic

**Comment by spuetz** [comment]
I'll give it a try ;)

**Comment by doisyg** [comment]
Can anybody confirm  that mbf is working under Melodic?

**Comment by corot** [comment]
Not yet, but I will install Melodic on next week and do so.

**Pull request merged**